Source/Core/Core/PowerPC/Expression.cpp

```diff
@@ -219,40 +220,161 @@ void ExprVarListDeleter::operator()(expr_var_list* vars) const
 Expression::Expression(std::string_view text, ExprPointer ex, ExprVarListPointer vars)
     : m_text(text), m_expr(std::move(ex)), m_vars(std::move(vars))
 {
+  using enum Expression::VarBindingType;
+  static const std::unordered_map<std::string_view, Expression::VarBinding> lookup = {
```

Comment:
Instead of making this an `unordered_map` which is always runtime-constructed, can you instead make this compile-time? `std::array<std::pair<std::string_view, Expression::VarBinding>>>`, `static_assert(std::is_sorted)`, `std::binary_search` should work just fine here.

Reply:
I like that suggestion, I will give it a go.

Reply:
`std::binary_search` just returns true if the given value exists in a sorted container, so I needed to write a new function for the job. It is found in the new `Common/Algorithm.h` header.

Reply:
The new `Common::Ranges::BinaryFind` algorithm wasn't inlining well, meaning some redundant conditions were being evaluated. Thus, I removed BinaryFind and the new header containing it.